File: 2837-minimum-operations-to-make-the-integer-zero/minimum-operations-to-make-the-integer-zero.py
class Solution:
    def makeTheIntegerZero(self, num1: int, num2: int) -> int:
        for op in range(1, 61, 1):
            diff = num1 - op * num2
            if diff < op: continue
            if diff.bit_count() <= op:
                return op

        return -1

File: 2837-minimum-operations-to-make-the-integer-zero/minimum-operations-to-make-the-integer-zero.cpp
class Solution {
    using ll = long long;
public:
    int makeTheIntegerZero(int num1, int num2) {
        for (int op = 1; op < 61; op++) {
            ll diff = 1LL * num1 - op * 1LL * num2;
            if (diff < op) continue;

            if (__builtin_popcountll(diff) <= op) return op;
        }

        return -1;
    }
};